File: php/get_subuser_traffic.php
<?php

// Information obtained from authentication

$userId = '';
$token = '';
$username = 'b35g5sg4h345'; // Name of your subuser
$query = array(
  'type' => '', // Available types: 24h, month, 7days, custom. If custom type is selected you must provide $from and $to parameters
  'from' => '', //yyyy-mm-dd
  'to' => ''
);
$url = 'https://api.smartproxy.com/v1/users/' . $userId . '/sub-users/' . $username . '/traffic?' . http_build_query($query, '', "&");

$ch = curl_init();

curl_setopt($ch, CURLOPT_URL, $url);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "GET");

$headers = array();
$headers[] = "Accept: application/json";
$headers[] = "Authorization: Token $token";
curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);

$result = curl_exec($ch);
if (curl_errno($ch)) {
    echo 'Error:' . curl_error($ch);
}
curl_close($ch);
echo $result;

File: csharp/get_subuser_traffic.cs
using System;
using System.IO;
using System.Net;

namespace Smartproxy_API
{
    class get_subuser_traffic
    {
        static void Main(string[] args)
        {
            // Details from authentication
            string userId = "";
            string token = "";

            string username = ""; // Subuser name
            string queryParam = ""; // Available types: 24h, month, 7days, custom. If custom type is selected you must provide from and to parameters.

            string uri = "https://api.smartproxy.com/v1/users/" + userId + "/sub-users/" + username + "/traffic?type=" + queryParam;

            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(uri);
            request.AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate;
            request.PreAuthenticate = true;
            request.Headers.Add("Authorization", "Token " + token);
            request.ContentType = "application/json";
            request.Method = "GET";

            using (HttpWebResponse response = (HttpWebResponse)request.GetResponse())
            using (Stream stream = response.GetResponseStream())
            using (StreamReader reader = new StreamReader(stream))
            {
                Console.WriteLine(reader.ReadToEnd());
                Console.ReadKey();
            }
        }
    }
}
